### ISSUE TITLE
CA1305 is invalid for Enum.ToString()

### ISSUE THREAD

**Issue by flcdrg** [dotnet/roslyn-analyzers#1072]
#### Analyzer package

System.Runtime.Analyzers
#### Analyzer

SpecifyIFormatProvider
#### Repro steps

``` C#
public enum Demo 
{
    One,
    Two
}

...

{
    Console.WriteLine(One.ToString());
}
```

the statement above raises CA1305.
#### Expected behavior

No warning raised
#### Actual behavior

If you try to fix the problem like this:

``` C#
{
    Console.WriteLine(One.ToString(CultureInfo.InvariantCulture));
}
```

Then CS0618 is raised "Enum.ToString(IFormatProvider)' is obsolete: 'The provider argument is not used. Please use ToString()."

That is correct - the overloaded ToString just calls ToString() internally so there is no point in calling ToString(IFormatProvider)

**Comment by mavasani** [comment]
This doesn't repro anymore.
